inc/endpoints/class-token.php

```diff
@@ -71,7 +76,7 @@ public function exchange_token( WP_REST_Request $request ) {
 			return $auth_code;
 		}
 
-		$is_valid = $auth_code->validate();
+		$is_valid = $auth_code->validate( $request );
```

Comment:
I'd rather pass the args in separately here to avoid having the `validate()` method depend on the request parameter names.